### PR TITLE
storage: always nil check resource group before deref

### PR DESCRIPTION
**Pull request by katbyte**

### CODE REVIEW

File: azurerm/resource_arm_storage_container.go [hunk 1]
```diff
@@ -135,10 +138,10 @@ func resourceArmStorageContainerUpdate(d *schema.ResourceData, meta interface{})
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group for Storage Account %q: %s", id.AccountName, err)
+		return fmt.Errorf("Error locating Resource Group for Storage Container %q (Account %s): %s", id.ContainerName, id.AccountName, err)
 	}
 	if resourceGroup == nil {
-		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Account %q - assuming removed & removing from state", id.AccountName)
+		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Container %q (Account %s)", id.ContainerName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_container.go]:
```suggestion
		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Container %q (Account %s) - assuming removed & removing from state", id.ContainerName, id.AccountName)
```

File: azurerm/resource_arm_storage_container.go [hunk 2]
```diff
@@ -184,10 +187,10 @@ func resourceArmStorageContainerRead(d *schema.ResourceData, meta interface{}) e
...
```
Comment (katbyte) [inline azurerm/resource_arm_storage_container.go]:
```suggestion
		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Container %q (Account %s) - assuming removed & removing from state", id.ContainerName, id.AccountName)
```

File: azurerm/resource_arm_storage_container.go [hunk 3]
```diff
@@ -239,10 +242,10 @@ func resourceArmStorageContainerDelete(d *schema.ResourceData, meta interface{})
...
```
Comment (katbyte) [inline azurerm/resource_arm_storage_container.go]:
```suggestion
		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Container %q (Account %s) - assuming removed & removing from state", id.ContainerName, id.AccountName)
```

File: azurerm/resource_arm_storage_queue.go [hunk 4]
```diff
@@ -139,11 +139,10 @@ func resourceArmStorageQueueRead(d *schema.ResourceData, meta interface{}) error
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group: %s", err)
+		return fmt.Errorf("Error locating Resource Group for Storage Container %q (Account %s): %s", id.QueueName, id.AccountName, err)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_queue.go]:
```suggestion
		return fmt.Errorf("Error locating Resource Group for Queue Container %q (Account %s): %s", id.QueueName, id.AccountName, err)
```

File: azurerm/resource_arm_storage_table.go [hunk 5]
```diff
@@ -205,11 +208,12 @@ func resourceArmStorageTableDelete(d *schema.ResourceData, meta interface{}) err
...
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group: %s", err)
+		return fmt.Errorf("Error locating Resource Group for Storage Table %q (Account %s): %s", id.TableName, id.AccountName, err)
 	}
 
 	if resourceGroup == nil {
-		log.Printf("Unable to determine Resource Group for Storage Account %q (assuming removed)", id.AccountName)
+		log.Printf("Unable to determine Resource Group for Storage Storage Table %q (Account %s)", id.TableName, id.AccountName)
+		d.SetId("")
```
Comment (mbfrahry) [inline azurerm/resource_arm_storage_table.go]: This is safe to remove
```suggestion
```

File: azurerm/resource_arm_storage_queue.go [hunk 6]
```diff
@@ -139,11 +139,10 @@ func resourceArmStorageQueueRead(d *schema.ResourceData, meta interface{}) error
...
 	}
-
 	if resourceGroup == nil {
-		log.Printf("[WARN] Unable to determine Resource Group for Storage Account %q (assuming removed) - removing from state", id.AccountName)
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Container %q (Account %s)", id.QueueName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_queue.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Queue %q (Account %s) - assuming removed & removing from state", id.QueueName, id.AccountName)
```

File: azurerm/resource_arm_storage_queue.go [hunk 7]
```diff
@@ -181,11 +180,10 @@ func resourceArmStorageQueueDelete(d *schema.ResourceData, meta interface{}) err
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group: %s", err)
+		return fmt.Errorf("Error locating Resource Group for Storage Container %q (Account %s): %s", id.QueueName, id.AccountName, err)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_queue.go]:
```suggestion
		return fmt.Errorf("Error locating Resource Group for Storage Queue %q (Account %s): %s", id.QueueName, id.AccountName, err)
```

File: azurerm/resource_arm_storage_table_entity.go [hunk 8]
```diff
@@ -169,10 +178,10 @@ func resourceArmStorageTableEntityDelete(d *schema.ResourceData, meta interface{
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group for Storage Account %q: %s", id.AccountName, err)
+		return fmt.Errorf("Error locating Resource Group for Storage Table %q (Account %s): %s", id.TableName, id.AccountName, err)
 	}
 	if resourceGroup == nil {
-		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Account %q - assuming removed already", id.AccountName)
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Table %q (Account %s)", id.TableName, id.AccountName)
 		d.SetId("")
```
Comment (mbfrahry) [inline azurerm/resource_arm_storage_table_entity.go]:
```suggestion
```

File: azurerm/resource_arm_storage_queue.go [hunk 9]
```diff
@@ -181,11 +180,10 @@ func resourceArmStorageQueueDelete(d *schema.ResourceData, meta interface{}) err
...
 	}
-
 	if resourceGroup == nil {
-		log.Printf("[WARN] Unable to determine Resource Group for Storage Account %q (assuming removed) - removing from state", id.AccountName)
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Container %q (Account %s)", id.QueueName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_queue.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Queue %q (Account %s) - assuming removed & removing from state", id.QueueName, id.AccountName)
```

File: azurerm/resource_arm_storage_share.go [hunk 10]
```diff
@@ -175,10 +178,10 @@ func resourceArmStorageShareRead(d *schema.ResourceData, meta interface{}) error
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group for Storage Account %q: %s", id.AccountName, err)
+		return fmt.Errorf("Error locating Resource Group for Storage Share %q (Account %s): %s", id.ShareName, id.AccountName, err)
 	}
 	if resourceGroup == nil {
-		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Account %q - assuming removed & removing from state", id.AccountName)
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Share %q (Account %s)", id.ShareName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_share.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Share %q (Account %s) - assuming removed & removing from state", id.ShareName, id.AccountName)
```

File: azurerm/resource_arm_storage_share.go [hunk 11]
```diff
@@ -234,10 +237,10 @@ func resourceArmStorageShareUpdate(d *schema.ResourceData, meta interface{}) err
...
```
Comment (katbyte) [inline azurerm/resource_arm_storage_share.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Share %q (Account %s) - assuming removed & removing from state", id.ShareName, id.AccountName)
```

File: azurerm/resource_arm_storage_share.go [hunk 12]
```diff
@@ -297,10 +300,10 @@ func resourceArmStorageShareDelete(d *schema.ResourceData, meta interface{}) err
...
```
Comment (katbyte) [inline azurerm/resource_arm_storage_share.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Share %q (Account %s) - assuming removed & removing from state", id.ShareName, id.AccountName)
```

File: azurerm/resource_arm_storage_share_directory.go [hunk 13]
```diff
@@ -106,7 +109,12 @@ func resourceArmStorageShareDirectoryUpdate(d *schema.ResourceData, meta interfa
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group: %s", err)
+		return fmt.Errorf("Error locating Resource Group for Storage Share Directory %q (Share %s, Account %s): %s", id.DirectoryName, id.ShareName, id.AccountName, err)
+	}
+	if resourceGroup == nil {
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Share Directory %q (Share %s, Account %s)", id.DirectoryName, id.ShareName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_share_directory.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Share Directory %q (Share %s, Account %s) - assuming removed & removing from state", id.DirectoryName, id.ShareName, id.AccountName)
```

File: azurerm/resource_arm_storage_share_directory.go [hunk 14]
```diff
@@ -132,10 +140,10 @@ func resourceArmStorageShareDirectoryRead(d *schema.ResourceData, meta interface
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group for Storage Account %q: %s", id.AccountName, err)
+		return fmt.Errorf("Error locating Resource Group for Storage Share Directory %q (Share %s, Account %s): %s", id.DirectoryName, id.ShareName, id.AccountName, err)
 	}
 	if resourceGroup == nil {
-		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Account %q - assuming removed & removing from state", id.AccountName)
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Share Directory %q (Share %s, Account %s)", id.DirectoryName, id.ShareName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_share_directory.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Share Directory %q (Share %s, Account %s) - assuming removed & removing from state", id.DirectoryName, id.ShareName, id.AccountName)
```

File: azurerm/resource_arm_storage_share_directory.go [hunk 15]
```diff
@@ -172,10 +180,10 @@ func resourceArmStorageShareDirectoryDelete(d *schema.ResourceData, meta interfa
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group for Storage Account %q: %s", id.AccountName, err)
+		return fmt.Errorf("Error locating Resource Group for Storage Share Directory %q (Share %s, Account %s): %s", id.DirectoryName, id.ShareName, id.AccountName, err)
 	}
 	if resourceGroup == nil {
-		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Account %q - assuming removed already", id.AccountName)
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Share Directory %q (Share %s, Account %s)", id.DirectoryName, id.ShareName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_share_directory.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Share Directory %q (Share %s, Account %s) - assuming removed already", id.DirectoryName, id.ShareName, id.AccountName)
```

File: azurerm/resource_arm_storage_table.go [hunk 16]
```diff
@@ -153,11 +156,11 @@ func resourceArmStorageTableRead(d *schema.ResourceData, meta interface{}) error
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group: %s", err)
+		return fmt.Errorf("Error locating Resource Group for Storage Table %q (Account %s): %s", id.TableName, id.AccountName, err)
 	}
 
 	if resourceGroup == nil {
-		log.Printf("Unable to determine Resource Group for Storage Account %q (assuming removed)", id.AccountName)
+		log.Printf("Unable to determine Resource Group for Storage Storage Table %q (Account %s)", id.TableName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_table.go]:
```suggestion
		log.Printf("Unable to determine Resource Group for Storage Storage Table %q (Account %s) - assuming removed & removing from state", id.TableName, id.AccountName)
```

File: azurerm/resource_arm_storage_table.go [hunk 17]
```diff
@@ -205,11 +208,12 @@ func resourceArmStorageTableDelete(d *schema.ResourceData, meta interface{}) err
...
```
Comment (katbyte) [inline azurerm/resource_arm_storage_table.go]:
```suggestion
		log.Printf("Unable to determine Resource Group for Storage Storage Table %q (Account %s) - assuming removed & removing from state", id.TableName, id.AccountName)
```

File: azurerm/resource_arm_storage_table.go [hunk 18]
```diff
@@ -237,11 +241,11 @@ func resourceArmStorageTableUpdate(d *schema.ResourceData, meta interface{}) err
...
```
Comment (katbyte) [inline azurerm/resource_arm_storage_table.go]:
```suggestion
		log.Printf("Unable to determine Resource Group for Storage Storage Table %q (Account %s) - assuming removed & removing from state", id.TableName, id.AccountName)
```

File: azurerm/resource_arm_storage_table_entity.go [hunk 19]
```diff
@@ -123,10 +132,10 @@ func resourceArmStorageTableEntityRead(d *schema.ResourceData, meta interface{})
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group for Storage Account %q: %s", id.AccountName, err)
+		return fmt.Errorf("Error locating Resource Group for Storage Table %q (Account %s): %s", id.TableName, id.AccountName, err)
 	}
 	if resourceGroup == nil {
-		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Account %q - assuming removed & removing from state", id.AccountName)
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Table %q (Account %s)", id.TableName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_table_entity.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Table %q (Account %s) - assuming removed & removing from state", id.TableName, id.AccountName)
```

File: azurerm/resource_arm_storage_table_entity.go [hunk 20]
```diff
@@ -169,10 +178,10 @@ func resourceArmStorageTableEntityDelete(d *schema.ResourceData, meta interface{
 
 	resourceGroup, err := storageClient.FindResourceGroup(ctx, id.AccountName)
 	if err != nil {
-		return fmt.Errorf("Error locating Resource Group for Storage Account %q: %s", id.AccountName, err)
+		return fmt.Errorf("Error locating Resource Group for Storage Table %q (Account %s): %s", id.TableName, id.AccountName, err)
 	}
 	if resourceGroup == nil {
-		log.Printf("[DEBUG] Unable to locate Resource Group for Storage Account %q - assuming removed already", id.AccountName)
+		log.Printf("[WARN] Unable to determine Resource Group for Storage Table %q (Account %s)", id.TableName, id.AccountName)
```
Comment (katbyte) [inline azurerm/resource_arm_storage_table_entity.go]:
```suggestion
		log.Printf("[WARN] Unable to determine Resource Group for Storage Table %q (Account %s) - assuming removed alreadu", id.TableName, id.AccountName)
```

**Pull request merged**